### PR TITLE
Openafs security updates

### PR DESCRIPTION
**Pull request by spacefrogg**
###### Motivation for this change
New packages for openAFS fixing security bugs.

The openafs-1.8.2 release also adds support for ARM64.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/servers/openafs/1.8/cross-build.patch [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+ 
+ AC_PROG_CC
+ AC_PROG_LIBTOOL
++AX_PROG_CC_FOR_BUILD
+ 
+ AC_PATH_PROGS([PATH_CPP], [cpp], [${CC-cc} -E], [$PATH:/lib:/usr/ccs/lib])
+ AC_SUBST([PATH_CPP])
+diff -Nur --unidirectional-new-file openafs-1.8.2/src/cf/ax_prog_cc_for_build.m4 openafs-1.8.2.new/src/cf/ax_prog_cc_for_build.m4
+--- openafs-1.8.2/src/cf/ax_prog_cc_for_build.m4	1970-01-01 01:00:00.000000000 +0100
++++ openafs-1.8.2.new/src/cf/ax_prog_cc_for_build.m4	2018-10-16 16:20:40.278641658 +0200
```
[the remark this answers is not in the document]
Comment (spacefrogg) [reply pkgs/servers/openafs/1.8/cross-build.patch]: This is a macro from the GNU Autoconf Archive (see also in the header of the file).

**Comment by Mic92** [comment]
For zfs we usually have no the upper bound if there is no known incompatibility and set it as build errors appear. This means in nixos unstable we sometimes propagate build errors to the users, if we are not marking incompatibilities fast enough. Not sure if it is the best approach. I suppose when somebody actively test the latest kernel with a third-party modules then disabling newer kernels proactively is better because the maintainer would test it before. When we do not have such a maintainer not having the upper bound is better because then the build errors become also apparent in hydra and we can fix it before a release.

File: pkgs/servers/openafs/1.8/cross-build.patch (hunk 1 above)
[the remark this answers is not in the document]
Comment (Mic92) [reply pkgs/servers/openafs/1.8/cross-build.patch]: Would `autoreconfHook` also update the file in question or is this something that openafs has vendored?

**Comment by spacefrogg** [comment]
I will update the PR, then, to not specify an upper bound and follow this approach when there is no known incompatibility at the time of the update.

File: pkgs/servers/openafs/1.8/cross-build.patch (hunk 1 above)
[the remark this answers is not in the document]
Comment (spacefrogg) [reply pkgs/servers/openafs/1.8/cross-build.patch]: As far as I know, this is something not normally distributed with autoconf but an additional archive of contributed macros.

**Comment by Mic92** [comment]
Somehow we ended up with a openafs reference in the manual.

**Comment by spacefrogg** [comment]
The error comes from the nixos client module in `nixos/modules/services/network-filesystems/openafs/client.nix`:
```
module = mkOption {
          default = config.boot.kernelPackages.openafs;
          type = types.package;
          description = "OpenAFS kernel module package. MUST match the userland package!";
        };
```
nix-instantiate does not like the `default` assignment anymore. Replacing the package name with `null` fixes this particular issue but destroys the semantics...

File: pkgs/servers/openafs/1.6/module.nix [hunk 2]
```diff
@@ -51,8 +51,8 @@ in stdenv.mkDerivation rec {
     license = licenses.ipl10;
     platforms = platforms.linux;
     maintainers = [ maintainers.z77z maintainers.spacefrogg ];
-    broken = versionOlder kernel.version "3.18" ||
-             versionAtLeast kernel.version "4.18";
+    broken = versionOlder kernel.version "3.18"
+             || stdenv.targetPlatform.isAarch64;
```
Comment (spacefrogg) [inline pkgs/servers/openafs/1.6/module.nix]: Okay, this line breaks the manual evaluation. But I don't know, why...

**Comment by Mic92** [comment]
I think there is also `defaultText` which is used instead of `default`.

File: pkgs/servers/openafs/1.6/module.nix (hunk 2 above)
Comment (Mic92) [reply pkgs/servers/openafs/1.6/module.nix]: Please keep this line as it is and set `defaultText` in the manual to fix evaluation.

**Comment by Mic92** [comment]
backport: cb996fdb60ae8495cae196758ce03b79e49b8ac0

**Pull request merged**